Show `vscode:prepublish` script output even if it failed

A simple script like `echo foobar` prints this:

```
> npx vsce package
npx: installed 67 in 4.988s
Executing prepublish script 'npm run vscode:prepublish'...

> sample-extension@0.0.1 vscode:prepublish /home/iliazeus/Repositories/sample-extension
> echo foobar

foobar
 DONE  Packaged: /home/iliazeus/Repositories/sample-extension/sample-extension-0.0.1.vsix (5 files, 2.04KB)
```

But if i change it to fail (`echo foobar && exit 1`), it prints this:

```
> npx vsce package              
npx: installed 67 in 4.731s
Executing prepublish script 'npm run vscode:prepublish'...
 ERROR  Command failed: npm run vscode:prepublish
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sample-extension@0.0.1 vscode:prepublish: `echo foobar && exit 1`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sample-extension@0.0.1 vscode:prepublish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/iliazeus/.npm/_logs/2020-04-17T07_44_46_912Z-debug.log
```

Notice that `foobar` is missing from the second output.

It makes it hard, for example, to debug prepublish script errors running in a CI pipeline.

```
iliazeus@iliazeus:~/Repositories/sample-extension/ > npm run vscode:prepublish

> sample-extension@0.0.1 vscode:prepublish /home/iliazeus/Repositories/sample-extension
> echo foobar && exit 1

foobar
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sample-extension@0.0.1 vscode:prepublish: `echo foobar && exit 1`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sample-extension@0.0.1 vscode:prepublish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/iliazeus/.npm/_logs/2020-04-17T08_12_41_892Z-debug.log
```

--- a/src/package.ts
+++ b/src/package.ts
@@ -20,7 +20,7 @@ import { getDependencies } from './npm';
 const readFile = denodeify<string, string, string>(fs.readFile);
 const unlink = denodeify<string, void>(fs.unlink as any);
 const stat = denodeify(fs.stat);
-const exec = denodeify<string, { cwd?: string; env?: any; maxBuffer?: number; }, { stdout: string; stderr: string; }>(cp.exec as any, (err, stdout, stderr) => [err, { stdout, stderr }]);
+const exec = denodeify<string, { cwd?: string; env?: any; maxBuffer?: number; }, { stdout: string; stderr: string; error: any }>(cp.exec as any, (error, stdout, stderr) => [undefined, { stdout, stderr, error }]);
 const glob = denodeify<string, _glob.IOptions, string[]>(_glob);
 
 const resourcesPath = path.join(path.dirname(__dirname), 'resources');
@@ -929,8 +929,10 @@ async function prepublish(cwd: string, manifest: Manifest, useYarn: boolean = fa
 
 	console.log(`Executing prepublish script '${useYarn ? 'yarn' : 'npm'} run vscode:prepublish'...`);
 
-	const { stdout, stderr } = await exec(`${useYarn ? 'yarn' : 'npm'} run vscode:prepublish`, { cwd, maxBuffer: 5000 * 1024 });
+	const { stdout, stderr, error } = await exec(`${useYarn ? 'yarn' : 'npm'} run vscode:prepublish`, { cwd, maxBuffer: 5000 * 1024 });
 	process.stdout.write(stdout);
+	// in case of error, stderr gets written by a top-level exception handler
+	if (error !== undefined) throw error;
 	process.stderr.write(stderr);
 }
 

